### PR TITLE
mutiny plus

### PR DESCRIPTION
**Pull request by futurepaul**

### CODE REVIEW

**Comment by futurepaul** [comment]
this is based on mutiny-node master rn but otherwise is pretty close I think

**Comment by TonyGiorgio** [comment]
Screenshot from 2023-07-07 12-13-02

There's no information about price on the main screen and this warning dialog makes it seem like they would immediately pay, but there's no pricing information. This warning about "are you sure" doesn't make sense because not enough information was given for them to make a decision.

**Comment by TonyGiorgio** [comment]
When I attempt to join when no balance, I get an `We do not have enough balance to pay the given amount.` error. 

We should tell them upfront that they need to fund the wallet first before paying.

**Comment by TonyGiorgio** [comment]
There's so much text here, would be nice if there's at least more indication and seperation around the price. It blends in to all the text. 

Screenshot from 2023-07-07 12-19-53

File: src/routes/settings/Plus.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+                        <Switch>
+                            <Match when={state.subscription_timestamp}>
+                                <img src={party} class="w-1/2 mx-auto" />
+                                <NiceP>
+                                    You're part of the mutiny! Enjoy the
+                                    following perks:
+                                </NiceP>
+                                <Perks alreadySubbed />
+                                <NiceP>
+                                    You'll get a renewal payment request on{" "}
```
Comment (TonyGiorgio) [inline src/routes/settings/Plus.tsx:116]:
```suggestion
                                    You'll get a renewal payment request around {" "}
```

**Comment by TonyGiorgio** [comment]
I like the `+` icon!  

Screenshot from 2023-07-07 12-23-39

File: src/routes/settings/Plus.tsx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    return (
+        <ul class="list-disc ml-8 font-light text-lg">
+            <Show when={props.alreadySubbed}>
+                <li>Smug satisfaction</li>
+            </Show>
+            <li>
+                Redshift <em>(coming soon)</em>
+            </li>
+            <li>
+                Easy gifting <em>(coming soon)</em>
```
Comment (TonyGiorgio) [inline src/routes/settings/Plus.tsx:31]:
```suggestion
                Gifting <em>(coming soon)</em>
```

I think just calling it gifting is better.

**Review by TonyGiorgio — CHANGES_REQUESTED** [review]
Some minor things to make sure this is solid and the edge cases are handled but everything else like restoring or expiration handling seems to work pretty well!

**Comment by futurepaul** [comment]
alright should be ready for another look

**Comment by TonyGiorgio** [comment]
Needs rebase.

File: src/components/Activity.tsx [hunk 3]
```diff
@@ -100,12 +109,13 @@ export function CombinedActivity(props: { limit?: number }) {
 
     createEffect(() => {
         if (!state.wallet_loading && !state.is_syncing) {
+            console.log("Syncing activity...");
```
Comment (TonyGiorgio) [inline src/components/Activity.tsx:112]: This can be removed, a bit spammy in the logs.

File: src/components/App.tsx [hunk 4]
```diff
@@ -23,12 +24,24 @@ export default function App() {
...
                 <header class="w-full flex justify-between items-center mt-4 mb-2">
                     <div class="flex items-center gap-2">
-                        <img
-                            id="mutiny-logo"
-                            src={pixelLogo}
-                            class="h-[25px] w-[75px]"
-                            alt="Mutiny logo"
-                        />
+                        <Switch>
+                            <Match when={state.subscription_timestamp}>
```
Comment (TonyGiorgio) [inline src/components/App.tsx:28]: I think there should be a function checking whether `state.subscription_timestamp` is not expired yet.

We can do a ~3 day grace period thing in a future PR later this month. But until then, all of the checks on this valud should also check whether or not it is in the past.

File: src/routes/settings/Plus.tsx [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+            console.error(e);
+        }
+    });
+
+    async function handleConfirm() {
+        try {
+            setSubbing(true);
+            setError(undefined);
+            // TODO: is it lame to hardcode this
+            const invoice = await state.mutiny_wallet?.subscribe_to_plan(0);
```
Comment (TonyGiorgio) [inline src/routes/settings/Plus.tsx:75]: it should just be `planDetails().id`

**Review by TonyGiorgio — CHANGES_REQUESTED** [review]
Testing works pretty well, just a few code things I noticed.

File: src/routes/settings/Plus.tsx [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+
+    return (
+        <MutinyWalletGuard>
+            <SafeArea>
+                <DefaultMain>
+                    <BackLink href="/settings" title="Settings" />
+                    <LargeHeader>Mutiny+</LargeHeader>
+                    <VStack>
+                        <Switch>
+                            <Match when={state.subscription_timestamp}>
```
Comment (TonyGiorgio) [inline src/routes/settings/Plus.tsx:181]: check expired

File: src/routes/settings/Plus.tsx [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+                                </NiceP>
+                                <NiceP>
+                                    To cancel your subscription just don't pay.
+                                    You can also disable the Mutiny+{" "}
+                                    <A href="/settings/connections">
+                                        Wallet Connection.
+                                    </A>
+                                </NiceP>
+                            </Match>
+                            <Match when={!state.subscription_timestamp}>
```
Comment (TonyGiorgio) [inline src/routes/settings/Plus.tsx:205]: check expired

**Review by TonyGiorgio — CHANGES_REQUESTED** [review]

**Comment by futurepaul** [comment]
nice catches ty

**Review by TonyGiorgio — APPROVED** [review]

**Comment by TonyGiorgio** [comment]
needs rebase

**Pull request merged**